### PR TITLE
[Concurrency] Add a few missing `Sendable` annotations in the concurrency library.

### PR DESCRIPTION
**Pull request by hborla**
* `ContinuousClock` and `SuspendingClock` are `Sendable`
* `Clock.measure` should inherit actor isolation.
* `Async{Throwing}Stream.init(unfolding:)` should accept a `@Sendable` closure.

Resolves: rdar://118444668, rdar://105405836

### CODE REVIEW

**Comment by hborla** [comment]
@swift-ci please smoke test

**Comment by hborla** [comment]
@swift-ci please smoke test

File: stdlib/public/Concurrency/Clock.swift [hunk 1]
```diff
@@ -67,7 +67,18 @@ extension Clock {
   ///          await someWork()
   ///       }
   @available(SwiftStdlib 5.7, *)
+  @_alwaysEmitIntoClient
   public func measure(
+    isolation: isolated (any Actor)? = #isolation,
```
Comment (hborla) [inline stdlib/public/Concurrency/Clock.swift:72]: This change will put `public func measure` behind `$OptionalIsolatedParameters` in the swiftinterface for the _Concurrency library, meaning the function will disappear when imported with a compiler that predates `OptionalIsolatedParameters`. I need to fix this before merging this change.

Comment (hborla) [reply stdlib/public/Concurrency/Clock.swift:73]: Fixed using `CONDITIONALLY_SUPPRESSIBLE_LANGUAGE_FEATURE`:

```
  #if compiler(>=5.3) && $ExpressionMacroDefaultArguments
  #if $OptionalIsolatedParameters
  @available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
  @_alwaysEmitIntoClient public func measure(isolation: isolated (any _Concurrency.Actor)? = #isolation, _ work: () async throws -> Swift.Void) async rethrows -> Self.Instant.Duration {
    try await measure(work)
  }
  #else
  @available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
  @_alwaysEmitIntoClient public func measure(isolation: (any _Concurrency.Actor)? = nil, _ work: () async throws -> Swift.Void) async rethrows -> Self.Instant.Duration {
    try await measure(work)
  }
  #endif
  #endif
```

**Comment by hborla** [comment]
@swift-ci please smoke test

**Comment by saagarjha** [comment]
Would you mind providing a bit more insight behind the changes to AsyncStream? This makes it annoying to wrap one stream in another, because AsyncStream's iterator is not Sendable and yet is expected to produce its results that way. Mentally I can kind of understand the former (don't share an iterator across multiple concurrency domains?) but I would assume AsyncStream should then inherit this too? Like, if I transform

```swift
for await _ in stream {
}
```

to

```swift
var iterator = stream.makeAsyncIterator()
while let await _ = iterator.next() {
}
```

Why can't the stream take a non-sendable generator that it unfolds for each call to next()?

**Pull request merged**